ManyWriteAheadLogsIT is flakey

While attempting to tune the timing of many of our ITs, I found that ManyWriteAheadLogsIT occassionally fails with an assertion error:

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 21.762 s <<< FAILURE! - in org.apache.accumulo.test.functional.ManyWriteAheadLogsIT
[ERROR] testMany(org.apache.accumulo.test.functional.ManyWriteAheadLogsIT)  Time elapsed: 21.538 s  <<< FAILURE!
java.lang.AssertionError: Open WALs not in expected range 0
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.apache.accumulo.test.functional.ManyWriteAheadLogsIT.addOpenWals(ManyWriteAheadLogsIT.java:167)
	at org.apache.accumulo.test.functional.ManyWriteAheadLogsIT.testMany(ManyWriteAheadLogsIT.java:140)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)

```

This occurs on the 1.9 branch, at the very least, but probably also affects the master/2.1 branch.

I ran the following to try to reproduce, expecting it to take some time:
```bash
while mvn clean verify -Dtest=blah -Dit.test=ManyWriteAheadLogsIT -Dspotbugs.skip -Dcheckstyle.skip; do echo Test passed...; done
```

It failed on the first attempt with:

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 17.589 s <<< FAILURE! - in org.apache.accumulo.test.functional.ManyWriteAheadLogsIT
[ERROR] testMany(org.apache.accumulo.test.functional.ManyWriteAheadLogsIT)  Time elapsed: 17.354 s  <<< FAILURE!
java.lang.AssertionError: Open WALs not in expected range 0
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.apache.accumulo.test.functional.ManyWriteAheadLogsIT.addOpenWals(ManyWriteAheadLogsIT.java:167)
	at org.apache.accumulo.test.functional.ManyWriteAheadLogsIT.testMany(ManyWriteAheadLogsIT.java:140)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)

```

It is possible that @jzgithub1 's suggestion to include `0` as a possible value in the test will "fix" the test, but without investigating further, I suspect, doing so would merely circumvent the test case the IT is looking for.

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
@@ -205,6 +205,8 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
       if (entry.getValue() == WalState.OPEN) {
         open++;
         allWalsSeen.add(entry.getKey());
+      } else {
+        log.error("The WalState is " + entry.getValue());// CLOSED or UNREFERENCED
       }
     }
 

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
@@ -201,15 +201,29 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
     Map<String,WalState> wals = WALSunnyDayIT._getWals(c);
     Set<Entry<String,WalState>> es = wals.entrySet();
     int open = 0;
-    for (Entry<String,WalState> entry : es) {
-      if (entry.getValue() == WalState.OPEN) {
-        open++;
-        allWalsSeen.add(entry.getKey());
-      } else {
-        log.error("The WalState is " + entry.getValue());// CLOSED or UNREFERENCED
+    int attempts = 1;
+    boolean foundWal = false;
+
+    while (open == 0) {
+      for (Entry<String,WalState> entry : es) {
+        if (entry.getValue() == WalState.OPEN) {
+          open++;
+          allWalsSeen.add(entry.getKey());
+          foundWal = true;
+        } else {
+          log.error("The WalState is " + entry.getValue());// CLOSED or UNREFERENCED
+        }
+
+        if (!foundWal) {
+          attempts++;
+          Thread.sleep(50);
+          if (attempts % 50 == 0)
+            log.info("Has not found an open WAL in " + attempts + " attempts.");
+        }
       }
     }
 
+    log.debug("It took " + attempts + " attempt(s) to find an open WAL");
     assertTrue("Open WALs not in expected range " + open, open > 0 && open < 4);
   }
 

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
@@ -213,7 +213,7 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
           allWalsSeen.add(entry.getKey());
           foundWal = true;
         } else {
-          log.error("The WalState is " + entry.getValue());// CLOSED or UNREFERENCED
+          log.debug("The WalState for {} is {}", entry.getKey(), entry.getValue()); // CLOSED or UNREFERENCED
         }
       }
 

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
@@ -220,7 +220,7 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
       if (!foundWal) {
         Thread.sleep(50);
         if (attempts % 50 == 0)
-          log.info("Has not found an open WAL in " + attempts + " attempts.");
+          log.debug("No open WALs found in {} attempts.", attempts);
       }
     }
 

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
@@ -198,13 +198,15 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
   }
 
   private void addOpenWals(ServerContext c, Set<String> allWalsSeen) throws Exception {
-    Map<String,WalState> wals = WALSunnyDayIT._getWals(c);
-    Set<Entry<String,WalState>> es = wals.entrySet();
+
     int open = 0;
-    int attempts = 1;
+    int attempts = 0;
     boolean foundWal = false;
 
     while (open == 0) {
+      attempts++;
+      Map<String,WalState> wals = WALSunnyDayIT._getWals(c);
+      Set<Entry<String,WalState>> es = wals.entrySet();
       for (Entry<String,WalState> entry : es) {
         if (entry.getValue() == WalState.OPEN) {
           open++;
@@ -213,13 +215,12 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
         } else {
           log.error("The WalState is " + entry.getValue());// CLOSED or UNREFERENCED
         }
+      }
 
-        if (!foundWal) {
-          attempts++;
-          Thread.sleep(50);
-          if (attempts % 50 == 0)
-            log.info("Has not found an open WAL in " + attempts + " attempts.");
-        }
+      if (!foundWal) {
+        Thread.sleep(50);
+        if (attempts % 50 == 0)
+          log.info("Has not found an open WAL in " + attempts + " attempts.");
       }
     }
 

--- a/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java
@@ -224,7 +224,7 @@ public class ManyWriteAheadLogsIT extends AccumuloClusterHarness {
       }
     }
 
-    log.debug("It took " + attempts + " attempt(s) to find an open WAL");
+    log.debug("It took {} attempt(s) to find {} open WALs", attempts, open);
     assertTrue("Open WALs not in expected range " + open, open > 0 && open < 4);
   }
 

